### ISSUE TITLE
TypeScript - Bad syntax generated for interface with no fields that extends another

### ISSUE THREAD

**Issue by brainlessdeveloper** [dotansimha/graphql-code-generator#1592]
**Describe the bug**
 

The schema I'm trying to generate types for contains some types that implement other types but don't contain any fields of their own. See the schema below for a minimal reproducible example.

The example below generates the following code:

```typescript
type Maybe<T> = T | null;
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string,
  String: string,
  Boolean: boolean,
  Int: number,
  Float: number,
};

export type GenericInterface = {};

export type SpecificInterface = GenericInterface & ;
```

The generated line `type SpecificInterface = GenericInterface & ;` is not valid TS.

**To Reproduce**

Run `graphql-codegen --config codegen.yml` with the config below. The code I've pasted is a minimal reproducible example and should work as of 1.0.6.

1. My GraphQL schema:

```json
{
  "__schema": {
    "queryType": null,
    "mutationType": null,
    "subscriptionType": null,
    "types": [
      {
        "kind": "INTERFACE",
        "name": "GenericInterface",
        "fields": [],
        "inputFields": null,
        "interfaces": null,
        "enumValues": null
      },
      {
        "kind": "OBJECT",
        "name": "SpecificInterface",
        "fields": [],
        "inputFields": null,
        "interfaces": [
          { "kind": "INTERFACE", "name": "GenericInterface", "ofType": null }
        ],
        "enumValues": null,
        "possibleTypes": null
      }
    ]
  }
}

```

2. My GraphQL operations:

Operations irrelevant.

3. My `codegen.yml` config file:

```yml
overwrite: true
schema: "./schema.json"
documents: "**/*.graphql"
generates:
  src/types.tsx:
    plugins:
      - "typescript"
      - "typescript-operations"
      - "typescript-react-apollo"
```

**Expected behavior**

Expected generated code:

```typescript
type Maybe<T> = T | null;
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string,
  String: string,
  Boolean: boolean,
  Int: number,
  Float: number,
};

export type GenericInterface = {};

export type SpecificInterface = GenericInterface & {};
```

I.e. the final line should be an intersection between the other types and `{}`, which represents no fields.

**Environment:**

```
    "@graphql-codegen/cli": "^1.0.6",
    "@graphql-codegen/typescript": "1.0.6",
    "@graphql-codegen/typescript-operations": "1.0.6",
    "@graphql-codegen/typescript-react-apollo": "1.0.6",
```

- NodeJS: v11.12.0

**Comment by dotansimha** [comment]
@brainlessdeveloper, thank you for reporting it. 
I managed to reproduce this issue and fix it: https://github.com/dotansimha/graphql-code-generator/pull/1596 

Can you please test the alpha version? `1.0.7-alpha-ba1a94bb.4`

**Comment by brainlessdeveloper** [comment]
How can I access the tag?

**Comment by dotansimha** [comment]
@brainlessdeveloper from npm, use that version instead of `1.0.6` in your `package.json` (just make sure to update all packages from `@graphql-codegen/...`)

**Comment by brainlessdeveloper** [comment]
Ah right, sorry :see_no_evil: 

The generated types look great now. Thanks!

**Comment by dotansimha** [comment]
@brainlessdeveloper Great, so I'll update it once we'll release a stable version with this fix.

**Comment by brainlessdeveloper** [comment]
@dotansimha Random question, but what do you use for tagging and versioning this repository? It seems pretty tidy.

**Comment by dotansimha** [comment]
@brainlessdeveloper We are using `lerna` for managing releases and tags, along with with CI configuration for auto-release of alpha versions automatically on each commit.
Changelog is done using GitHub Releases (on tags that have been created by `lerna`). 
We tried https://github.com/conventional-changelog/conventional-changelog in the past but it created a lot of clutter on the changelogs, so we prefer to do it manually.

**Comment by brainlessdeveloper** [comment]
Thanks a lot!

**Comment by dotansimha** [comment]
Fixed in 1.0.7 :)
